src/types/index.tsx

```diff
@@ -0,0 +1,38 @@
+export interface Todo {
+  id: number,
+  userId: number,
+  title: string,
+  completed: boolean,
+  createdAt: string,
+  updatedAt: string,
+}
+
+export type Todos = {
+  todos: Todo[],
+};
+
+export interface User {
+  id?: number,
+  name?: string,
+  username?: string,
+  email?: string,
+  phone?: string,
+  website?: string,
+  createdAt?: string,
+  updatedAt?: string,
+}
+
+export type RootState = {
+  todos: Todo []|[];
+  user: User | Record<string|number, never>;
```

Comment:
![image](https://user-images.githubusercontent.com/38065883/129341000-4e7a311c-397c-4e76-8034-c5a252d8b263.png)
```suggestion
  user: User | null;
```
Думаю такого типа будет достаточно